src/Microsoft.OData.Client/Materialization/FeedAndEntryMaterializerAdapter.cs

```diff
@@ -296,6 +323,87 @@ private ODataResourceSet ReadFeedCore(bool lazy)
             return result;
         }
 
+        /// <summary>
+        /// Reads an <see cref="ODataDeltaResourceSet"/>.
+        /// </summary>
+        /// <returns>The <see cref="MaterializerDeltaFeed"/> of the read <see cref="ODataDeltaResourceSet"/>.</returns>
+        private MaterializerDeltaFeed ReadDeltaFeedCore()
+        {
+            this.ExpectState(ODataReaderState.DeltaResourceSetStart);
+
+            ODataDeltaResourceSet result = (ODataDeltaResourceSet)this.reader.Item;
+            MaterializerDeltaFeed feed = null;
+
+            if (result == null)
+            {
+                return null;
+            }
+
+            feed = MaterializerDeltaFeed.CreateDeltaFeed(
+                    result,
+                    new List<IMaterializerState>(),
+                    this.materializerContext);
+
+            if (this.items?.Count > 0)
+            {
+                this.items.Push(feed);
+            }
+            else
+            {
+                IMaterializerState topLevelItem = feed;
+                this.items.Push(topLevelItem);

```

Comment:
@ElizabethOkerio The statement `if (this.items?.Count > 0)` suggests `this.items` can be null. We would end up in this block if that was the case. Would `this.items.Push(topLevelItem)` not throw an exception?

Reply:
fixed